app/Http/Controllers/Forum/TopicsController.php

```diff
@@ -160,6 +160,13 @@ public function reply(HttpRequest $request, $id)
             'body' => 'required',
         ]);
 
+        $trimmedBody = trim(Request::input('body'));
+
+        if (preg_match('/^\[quote(?:=".+?")?\].+?\[\/quote\]/s', $trimmedBody, $matches)
```

Comment:
should be done in model validation (the body validation above is outdated and shouldn't be there either).